File: 백준/Bronze/7770. 아즈텍 피라미드/아즈텍 피라미드.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int blocks = 0;
        int x = 0;
        int y = 1;
        int cnt = 0;
        while (true) {
            blocks += (x * x + y * y);
            if (n < blocks) {
                break;
            }
            x += 1;
            y += 1;
            cnt += 1;
        }
        System.out.println(cnt);
    }
}

File: 백준/Bronze/7770. 아즈텍 피라미드/아즈텍 피라미드.c
#include <stdio.h>

int main() {
    long n;
    scanf("%ld", &n);  
    long blocks = 0;
    long x = 0;
    long y = 1;  
    int cnt = 0;
    while (1) {
        blocks += (x * x + y * y);
        if (n < blocks) {
            break;
        }
        x += 1;
        y += 1;
        cnt += 1;
    }
    printf("%d\n", cnt);
    return 0;
}